### PR TITLE
fix: check for empty search #2433

### LINKED ISSUE #2433: Clear Filter gets highlighted on No Param Passes
**Describe the bug**

When there are no parameters in search bar on startup page and After clicking search , Clear Filter gets highlighted .

**To Reproduce**

1. Go to 'Start-Up Page'
2. Click on 'Search'

**Expected behavior**

Unless some parameter is passed ,Clear-Filter should not get highlighted.

**Screenshots**

Screenshot 2019-03-31 at 5 15 44 AM

**Desktop (please complete the following information):**
 - OS: Mac OS Mojave
 - Browser: Chrome
 - Version: 73.0.3

### PR DESCRIPTION
**Pull request by ridhishjain**
Fixes #2433  

#### Short description of what this resolves:
The clear filter button on the landing page won't get highlighted when an empty search is triggered.
But even if one of the three filters are applied, it will get highlighted on clicking search
ezgif com-crop (1)

#### Changes proposed in this pull request:

Initially when something was written in the event name field and search was clicked, the clear filter option was highlighted
When the filters were cleared, and empty search was triggered, clear filter was not highlighted
Similarly, when date filter was selected, it got highlighted on clicking search

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests, and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/quick-filter.js [hunk 1]
```diff
@@ -22,28 +20,48 @@ export default Component.extend({
...
+          .toISOString();
+        newEndDate = moment()
+          .endOf('week')
+          .toISOString();
         break;
 
       case 'this_weekend':
-        newStartDate = moment().isoWeekday('Friday').toISOString();
-        newEndDate = moment().isoWeekday('Sunday').toISOString();
+        newStartDate = moment()
```
Comment (kushthedude) [inline app/components/quick-filter.js]: Why ?

File: app/components/quick-filter.js [hunk 2]
```diff
@@ -57,7 +75,9 @@ export default Component.extend({
     this._super(...arguments);
     this.set('dummyLocation', this.location);
     this.set('dummyName', this.eventName);
-    if (this.dummyName || this.dummyLocation) {this.set('disableClear', false)}
+    if (this.dummyName || this.dummyLocation) {
```
Comment (kushthedude) [inline app/components/quick-filter.js]: Why ?

File: app/components/quick-filter.js [hunk 3]
```diff
@@ -12,7 +11,6 @@ export default Component.extend({
     let newEndDate = null;
 
     switch (this.filterDate) {
-
```
Comment (kushthedude) [inline app/components/quick-filter.js]: Revert

File: app/components/quick-filter.js [hunk 4]
```diff
@@ -2,7 +2,6 @@ import Component from '@ember/component';
 import moment from 'moment';
 
 export default Component.extend({
-
```
Comment (kushthedude) [inline app/components/quick-filter.js]: Revert

File: app/components/quick-filter.js (hunk 2 above)
Comment (ridhishjain) [reply app/components/quick-filter.js]: This change and all the above ones are occurring automatically on saving, how can I fix this?

**Comment by kushthedude** [comment]
Change your IDE or stop beautifying in IDE

File: app/components/quick-filter.js [hunk 5]
```diff
@@ -57,7 +57,7 @@ export default Component.extend({
     this._super(...arguments);
     this.set('dummyLocation', this.location);
     this.set('dummyName', this.eventName);
-    if (this.dummyName || this.dummyLocation) {this.set('disableClear', false)}
+    if (this.dummyName || this.dummyLocation) {this.set('disableClear', false);}
```
Comment (kushthedude) [inline app/components/quick-filter.js]: Why

**Comment by kushthedude** [comment]
Give a gif of the working functionality after the change

**Comment by ridhishjain** [comment]
@kushthedude review please.

**Comment by ridhishjain** [comment]
I looked at the closed PR related to this issue
[here](https://github.com/fossasia/open-event-frontend/pull/3548)

I changed my code according to that approach, filters are now working properly

**Comment by kushthedude** [comment]
> here

Is date filter working ?

**Comment by ridhishjain** [comment]
Suppose the 'event name' and 'location' field are empty, and date filter is selected 'this_week'. Then on clicking search, clear filter should get highlighted ??

**Comment by kushthedude** [comment]
Yes see the current behaviour

**Comment by ridhishjain** [comment]
Now, the current Scenario is if any one of the filters is selected or given some value, on-search, the clear filter will get highlighted. When all of them are left as blank, it won't get highlighted.
Date filter, location filter, and name filter are all working fine.

GIF is also updated

**Comment by ridhishjain** [comment]
@kushthedude review please

**Comment by iamareebjamal** [comment]
Which issue does this fix. Can't see the issue so no idea what behaviour to expect

**Comment by ridhishjain** [comment]
> Which issue does this fix. Can't see the issue so no idea what behaviour to expect

it fixes issue#2433

**Comment by kushthedude** [comment]
PR is abandoned and fork is deleted too.

**Pull request closed without merging**